src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int INCREASE_IN_TWO_TIMES = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
 
+        if (key == null) {
+            putForNullKey(value);
+        } else if (table[hash(key)] == null) {
+            putOnEmptyBucket(key, value);
+        } else {
+            putWithCollision(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return table[0] != null ? table[0].value : null;
+        }
+
+        int index = hash(key);
+        Node<K, V> nodeInBucket = table[index];
+        while (nodeInBucket != null) {
+            if (key.equals(nodeInBucket.key)) {
+                return nodeInBucket.value;
+            }
+            nodeInBucket = nodeInBucket.nextNode;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> nextNode;
+
+        public Node(K key, V value, Node<K, V> nextNode) {
+            this.key = key;
+            this.value = value;
+            this.nextNode = nextNode;
+        }
+    }
+
+    public int hash(K key) {
```

Comment:
It's not hash, it's bucket index calculated from hashcode
```suggestion
    public int calculateIndexByHashcode(K key) {
```